tests/wpunit/CommentObjectQueriesTest.php

```diff
@@ -914,4 +914,290 @@ static function ( $text, $comment ) use ( $filtered ) {
 		$this->assertArrayNotHasKey( 'errors', $actual );
 		$this->assertSame( apply_filters( 'comment_text', $filtered, null ), $actual['data']['comment']['content'] );
 	}
+
+	public function testDirectCommentQueries() {
+
+		$post_id = $this->factory()->post->create(
+			[
+				'post_content' => 'Post object',
+				'post_author'  => $this->admin,
+				'post_status'  => 'publish',
+			]
+		);
+
+		$comment_id = self::factory()->comment->create([
+			'comment_post_ID' => $post_id,
+			'comment_content' => 'Test direct comment query',
+			'comment_approved' => '1',
+		]);
+	
+		$query = '
+		query GetComment($id: ID!) {
+			comment(id: $id) {
+				id
+				content
+				status
+				date
+				author {
+					node {
+						__typename
+					}
+				}
+			}
+		}
+		';
+	
+		$variables = [
+			'id' => \GraphQLRelay\Relay::toGlobalId('comment', $comment_id),
+		];
+	
+		$actual = $this->graphql(['query' => $query, 'variables' => $variables]);
+	
+		$this->assertArrayNotHasKey('errors', $actual);
+	
+		wp_delete_post( $post_id, true );
+		wp_delete_comment( $comment_id );
+	
+		$this->assertEquals( apply_filters( 'comment_text', 'Test direct comment query' ), $actual['data']['comment']['content']);
+		$this->assertEquals('APPROVE', $actual['data']['comment']['status']);
+		
+	}
+
+	public function testCommentStatusFiltering() {
+		wp_set_current_user($this->admin); // Ensure we have permission to see all comments
+	
+		$post_id = $this->factory()->post->create(
+			[
+				'post_content' => 'Post object',
+				'post_author'  => $this->admin,
+				'post_status'  => 'publish',
+			]
+		);
+		
+		// Create approved comment
+		$approved_id = $this->factory()->comment->create([
+			'comment_post_ID' => $post_id,
+			'comment_content' => 'Approved comment',
+			'comment_approved' => 1,
+		]);
+	
+		// Create pending comment
+		$spam_comment_id = $this->factory()->comment->create([
+			'comment_post_ID' => $this->post_id,
+			'comment_content' => 'Pending comment',
+			'comment_approved' => 'spam',
+		]);
+	
+		$query = '
+		query GetComments($status: [CommentStatusEnum]) {
+			comments(where: { statusIn: $status }) {
+				nodes {
+					id
+					content
+					status
+				}
+			}
+		}
+		';
+	
+		// Test APPROVE status
+		$variables = ['status' => [ 'APPROVE' ] ];
```

Comment:
We probably want a test case with multiple status too.

Reply:
done!